### PR TITLE
Validate the uniqueness of email address

### PR DESCRIPTION
**Pull request by tuzz**
We have a unique index on the column, but the
error messages are friendlier if we add the
validation in as well.

### CODE REVIEW

File: spec/models/subscriber_spec.rb [hunk 1]
```diff
@@ -41,5 +41,19 @@
...
+
+      subject.address = "foo@bar.com"
+      expect(subject).to be_invalid
+    end
+
+    it "is case-insensitive when validating if an address is taken" do
+      FactoryGirl.create(:subscriber, address: "FOO@bar.com")
+
+      subject.address = "foo@BAR.COM"
+      expect(subject).to be_invalid
```
Comment (thomasleese) [inline spec/models/subscriber_spec.rb]: Technically, these are distinct email addresses. The part before the @ is case sensitive, the part after the @ is not.

Comment (thomasleese) [reply spec/models/subscriber_spec.rb]: (massive edge case, and most servers tend to be case insensitive on the part before @ too, so maybe it doesn't matter)

Comment (tuzz) [reply spec/models/subscriber_spec.rb]: Ahh – I didn't know that. I think we shouldn't restrict it in that case. It's simpler without, anyway.

Comment (thomasleese) [reply spec/models/subscriber_spec.rb]: I suppose we could write our own validation which checks the part after the @ being case sensitive and the part before being case insensitive, but that feels a bit extreme.

**Pull request merged**